qtfred/src/mission/dialogs/MissionGoalsDialogModel.cpp

```diff
@@ -11,50 +11,43 @@ namespace dialogs {
 MissionGoalsDialogModel::MissionGoalsDialogModel(QObject* parent, fso::fred::EditorViewport* viewport) :
 	AbstractDialogModel(parent, viewport) {
 }
-bool MissionGoalsDialogModel::apply() {
-	char buf[256], names[2][MAX_GOALS][NAME_LENGTH];
-	int i, count;
-
-	for (i=0; i<Num_goals; i++)
+bool MissionGoalsDialogModel::apply()
+{
+	for (int32_t i = 0; i < Num_goals; i++)
```

Comment:
Why are you using `int32_t` instead of `int` or `auto` or `size_t`?

Reply:
I am trying to use fixed width integers wherever possible to make the size consistent.

Reply:
But this is a `for` loop. (As are the other instances.) It's not a struct or a memory map. Why should that matter?

Reply:
To be honest, that is just the coding standard of my work so I used it out of habit. However, I think it is a worthwhile endeavour to use fixed with types wherever possible so I would prefer to leave it like it is.

Reply:
I wouldn't normally care, but for two reasons:

1) You've been proactive about using `auto` in many other places
2) There have only ever been a maximum of 5 goals.  A three-bit data type would be sufficient.  (The dialog has room for 10 so you could go to a whopping four bits.)  Even a `short` would be suitable.  But you're using a 32-bit type.

Reply:
I guess I could use auto instead although that is even longer than `int`...